### PR TITLE
Publish course validation

### PR DESCRIPTION
**Pull request by defong**
### Context
User can only publish a course if it passes validation

### Changes proposed in this pull request
User can only publish a course if it passes validation

### Guidance to review
sal val pub
fee val pub

### CODE REVIEW

**Comment by dankmitchell** [comment]
All good from a frontend/ui pov 💪

**Comment by defong** [comment]
[Validate 'About your course' mandatory fields on publish](https://trello.com/c/TCcBjbUi/125-validate-about-your-course-mandatory-fields-on-publish)

**Comment by defong** [comment]
[Course publish validation doesn't consider course duration and salary length](https://trello.com/c/uO01nOCT/200-course-publish-validation-doesnt-consider-course-duration-and-salary-length)

File: src/ui/ViewModels/FromUcasViewModel.cs [hunk 1]
```diff
@@ -36,15 +35,15 @@ public class CourseVariantViewModel
...
               : string.Equals("not running", Status ?? "", StringComparison.InvariantCultureIgnoreCase) ? CourseVariantStatus.NotRunning
-              : CourseVariantStatus.New;            
-        
+              : CourseVariantStatus.New;
+
         public IEnumerable<SchoolViewModel> Schools { get; set; }
 
     }
-    
+
```
Comment (MartinBentley) [inline src/ui/ViewModels/FromUcasViewModel.cs]: We need to think about splitting out this file. It contains 4 separate entities!

Comment (defong) [reply src/ui/ViewModels/FromUcasViewModel.cs]: carded up

https://trello.com/c/lN9Z51QU/235-tidy-up

**Pull request merged**